measure

```diff
@@ -9,86 +9,112 @@ import yaml
+
+        metrics = config["metrics"]
+        assert isinstance(metrics, dict), "The `metrics` attribute in `{}` section must be a dict".format(driver_key)
+        for m, mdesc in config["metrics"].items():
+            assert isinstance(m, str), "Metric name `{}` must be a string, found {}".format(m, type(m).__name__)
             if not isinstance(mdesc, dict):
-                raise Exception('Metric `{}` data in file {} must be a dict, found {}'.
-                    format(m, path, type(mdesc).__name__))
-            bad_keys = mdesc.keys() - {'query', 'unit'}
+                raise Exception(
+                    "Metric `{}` data in file {} must be a dict, found {}".format(
+                        m, path, type(mdesc).__name__
+                    )
+                )
+            bad_keys = mdesc.keys() - {"query", "unit"}
             if bad_keys:
-                raise Exception('Unknown attributes for metric `{}` in `prom` section of file {}: {}'.
-                    format(m, path, bad_keys))
-            if 'query' not in mdesc:
-                raise Exception('Missing `query` attribute for metric `{}` in `prom` section of file {}'.format(m, path))
-            if not isinstance(mdesc['query'], str):                
-                raise Exception('Query for metric `{}` must be string, found {}'.format(m, type(mdesc['query']).__name__))
-
-        return config
-
-    def get_prom_metrics(self):
-        '''
-        Get metrics known to Prometheus's endpoint
-        (Currently not used)
-        '''
-        base_url = PROM_BASE_URL
-
-        url = '%s%slabel/__name__/values'%(base_url, PROM_API_PATH)
-
-        self.debug("Getting url: ", url)
-        r = requests.get(url)
-
-        assert r.status_code == 200, \
-            'Prometheus server returned http code: ' + str(r.status_code)
+                raise Exception(
+                    "Unknown attributes for metric `{}` in `{}` section of file {}: {}".format(driver_key, m, path,
+                                                                                               bad_keys))
+            if "query" not in mdesc:
+                raise Exception(
+                    "Missing `query` attribute for metric `{}` in `{}` section of file {}".format(
+                        driver_key, m, path
+                    )
+                )
+            if not isinstance(mdesc["query"], str):
+                raise Exception(
+                    "Query for metric `{}` must be string, found {}".format(
+                        m, type(mdesc["query"]).__name__
+                    )
+                )
+
+            return config
+
+        def get_prom_metrics(self):
+            """
+            Get metrics known to Prometheus's endpoint
+            (Currently not used)
+            """
+            base_url = self.get_config().get("prometheus_endpoint", PROM_BASE_URL).rstrip("/")
+
+            url = "%s%slabel/__name__/values" % (base_url, PROM_API_PATH)
```

Comment:
I believe we decided to ditch `%` approach company-wide

Reply:
This is a part of the original driver, not changing it.